plugins/plugin-site/src/components/PluginReleases.jsx

```diff
@@ -37,12 +37,15 @@ function PluginReleases({pluginId, versions}) {
     return (
         <div id="pluginReleases--container" className="container">
             {versions.map(version => {
-                const release = releases.find(release => version.version === release.tagName.replace(/^[^0-9]*/, '')) || {};
+                console.log(version.version);
+                // assume tags are in the form ${pluginId}${infix}${version} or ${prefix}${version}
+                // where pluginId may contain digits, but prefix and infix cannot and version always starts with a digit
+                const release = releases.find(release => version.version === release.tagName.replace(pluginId, '').replace(/^[^0-9]*/, '')) || {};
                 return (
                     <div key={version.id} className="item card">
                         <div className="card-header">
                             <h5 className="card-title d-flex justify-content-between">
-                                <div><a href={version.url}>{release.name || version.version}</a></div>
+                                <div><a href={version.url}>{release.name && release.name != release.tagName ? release.name : version.version}</a></div>
```

Comment:
Only use the release name if it's human friendly, e.g. the plain version number `0.14` is prefered over `kubernetes-credentials-provider-0.14` as listed in https://github.com/jenkinsci/kubernetes-credentials-provider-plugin/releases

Reply:
ternaries tend to be a code smell for me. Can you pull it out into a tiny function for readability?